### DISCUSSION TITLE
Font choice and saving heatmaps with custom dimension

### DISCUSSION THREAD

**Discussion by timz0605** [david-barnett/microViz#164] [Q&A]
Hello David,

Starting a new discussion post here for another aesthetics-related question and also asking a minor question.

**1/ Is there a way to use custom fonts and adjust the aesthetics in figures created with microviz?**

In `ord_plot`, I changed the font to what I want by adding the code `theme(text = element_text(family = "Georgia")` since it is ggplot based and I am more familiar with it. But for heatmaps, I was wondering if there is a way to choose custom fonts? Also, can I adjust other aesthetics such as font size in heatmaps?

**2/ What is the recommended way of saving a heatmap, if I want more customization?**

for ggplot-based figures, I always use `ggsave()` since I can customize the `dpi` I want and also the dimension freely. But for heatmaps, I am a bit unsure how I can achieve the same result.

Thank you so much!

**Comment by david-barnett** [comment]
The heatmaps are indeed not using ggplot - so all styling and saving of cor_heatmap or comp_heatmap is based on standard approaches for ComplexHeatmap output

For fonts:
https://jokergoo.github.io/ComplexHeatmap-reference/book/integrate-with-other-packages.html#gridtext

For saving and styling, same methods as with base `plot()` graphics I believe. Open a graphics device (must be raster e.g. `png()` ) with a certain size and resolution and filepath, draw the plot, then close the graphics device.

Or if using rmarkdown or quarto, set the chunk fig width and fig height and dpi

**Reply by david-barnett** [reply]
for more info: [check https://bookdown.org/rdpeng/exdata/graphics-devices.html](https://bookdown.org/rdpeng/exdata/graphics-devices.html#how-does-a-plot-get-created)
